clamnibs/beamformer.py

```diff
@@ -200,14 +202,15 @@ def set_forward(raw, l_freq_noise, h_freq_noise, n_comp=4):
         w = _get_lcmv_weights(COV_B, M[:, ix_comp])
         psd, freqs = psd_array_welch(
             w @ data_broad, raw.info['sfreq'], fmin=1, fmax=40, n_fft=int(3 * raw.info['sfreq']))
-        axes[ix_comp, 0].semilogy(freqs, psd, c='k')
+        axes[ix_comp, 0].semilogy(freqs, psd.flatten(), c='k')
         axes[ix_comp, 0].tick_params(axis='x', labelsize=8)
         axes[ix_comp, 0].tick_params(axis='y', labelsize=5)
         axes[ix_comp, 0].axvline(
             l_freq_target, color='grey', linestyle='--', linewidth=0.5)
         axes[ix_comp, 0].axvline(
             h_freq_target, color='grey', linestyle='--', linewidth=0.5)
-        plot_topomap(M[:, ix_comp], raw.info, axes=axes[ix_comp,
+        
+        plot_topomap(M[:, ix_comp], mne.pick_info(raw.info, ixs_goods), axes=axes[ix_comp,
```

Comment:
Strange that you needed to use `pick_info `here, normally `plot_topomap `does this internally (because the bad channels are marked in raw.info).

Reply:
OK, let's leave this then.